code/game/objects/items/mail.dm

```diff
@@ -332,3 +334,166 @@
 /obj/item/paper/fluff/junkmail_generic/Initialize(mapload)
 	default_raw_text = pick(GLOB.junkmail_messages)
 	return ..()
+
+// =====================
+// Antag stuff down here
+// =====================
+
+/obj/item/mail/traitor
+	var/armed = FALSE
+	var/mob/madeby
+	goodie_count = 0
+
+/obj/item/mail/traitor/envelope
+	name = "envelope"
+	icon_state = "mail_large"
+	stamp_max = 2
+	stamp_offset_y = 5
+
+/obj/item/mail/traitor/attack_self(mob/user)
+	if(recipient_ref)
+		var/datum/mind/recipient = recipient_ref.resolve()
+		// If the recipient's mind has gone, then anyone can open their mail
+		// whether a mind can actually be qdel'd is an exercise for the reader
+		if(recipient && recipient != user?.mind)
+			to_chat(user, span_notice("You can't open somebody else's mail! That's <em>illegal</em>!"))
+			return
+
+	to_chat(user, span_notice("You start to unwrap the package..."))
+	if(!do_after(user, 1.5 SECONDS, target = user))
+		return
+	user.temporarilyRemoveItemFromInventory(src, TRUE)
+	playsound(loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+	for(var/obj/item/stuff in contents) // Mail and envelope actually can have more than 1 item.
+		if(user.put_in_hands(stuff) && armed)
+			log_bomber(user, "opened armed mail made by [madeby] ([madeby.ckey]), activating", stuff)
+			stuff.attack_self(user)
+	qdel(src)
+
+/obj/item/mail/traitor/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(armed)
+		if(W.tool_behaviour == TOOL_MULTITOOL)
+			to_chat(user, span_notice("You start to disable something..."))
+			
+			if(do_after(user, 2 SECONDS, target = src))
+				to_chat(user, span_notice("You have disarmed some kind of device..."))
+				playsound(src, 'sound/machines/defib_ready.ogg', 50, TRUE)
+				armed = FALSE
+
+/obj/item/storage/mail_counterfeit_device
+	name = "mail counterfeit device"
+	desc = "Device that actually able to counterfeit NT's mail. This device also able to place a trap inside of mail for malicious actions. Trap will \"activate\" any item inside of mail. Also it might be used for contraband purposes. Integrated micro-computer will give you great configuration optionality for your needs."
+	w_class = WEIGHT_CLASS_NORMAL
+	icon = 'icons/obj/device_syndie.dmi'
+	icon_state = "mail_counterfeit_device" // placeholder for now
+
+/obj/item/storage/mail_counterfeit_device/Initialize(mapload)
+	. = ..()
+	atom_storage.max_slots = 1
+	atom_storage.allow_big_nesting = TRUE
+	atom_storage.max_specific_storage = WEIGHT_CLASS_NORMAL
+
+/obj/item/storage/mail_counterfeit_device/attack_self(mob/user, modifiers)
+	var/mail_type = tgui_alert(user, "Is it gonna be an envelope or a normal mail?", "Mail Counterfeiting", list("Mail", "Envelope"))
+	if(isnull(mail_type))
+		return FALSE
+	if(!(src in user.contents))
+		return FALSE
+	mail_type = lowertext(mail_type)
```

Comment:
Why bother?

Reply:
Because it would be bad, if this device will fall from your hands and still be in process of creation.

Reply:
Why are we bothering with the lowertext?

Reply:
Ah... Because i want to see capitalazed text in tgui_alert, but for naming (because items are being named without capitalization) purpuse it has to be lowertext'ed.

Reply:
Ok, I see